samples/js/photos/src/main.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {searchToParams} from './dom.js';
+import Nav from './nav.js';
+import PhotosPage from './photos-page.js';
+import Viewport from './viewport.js';
+import {createPhoto} from './photo.js';
+import type {PhotoIndex, NomsPhoto} from './types.js';
+import {Path, PathSpec, Struct} from '@attic/noms';
+
+// Cache of index paths to indices. Otherwise calls to render are pretty slow,
+// which is noticeable when resizing, toggling between full screen photos, etc.
+const indexMap: Map<string, PhotoIndex> = new Map();
```

Comment:
Why is that? Is it actually going back to http each time? I thought there was an in-memory cache?